### DISCUSSION TITLE
Mirroring/flipping circuits

### DISCUSSION THREAD

**Discussion by aquamiau** [flandreas/antares#619] [Ideas]
As suggested in #285, having "Flip horizontal" and "Flip vertical" commands to mirror individual elements or entire selections would be useful for managing layouts.
While most if not all elements already have mirror properties, they are often disguised in other property names (for example, the Tri-State Buffer has "Control Input Orientation" or the Splitter "Position Bit 0") - flipping commands would also save someone from searching the right property that mirrors the element.

**Comment by flandreas** [comment]
Well, I wouldn't agree that most if not all elements have mirror properties. In fact, almost none besides the custom subcircuit symbols have. Take all logic gates, all switches, LEDs etc. None of them have a mirror property. And don't confuse it with the "Orientation" property: This is effectively rotation, which is much easier to implement, because it can be done "on the fly", as the underlying graphics engine supports rotation. The only thing you have to then do is rotating user input events to adjust for the rotated component. For mirroring, none of this exists for free.

Yes, some components like the Tri-State Buffer have properties like "Control Input Orientation" that can be seen as horizontal mirroring. But all of these had to be programmed in a special way. I currently don't know of a way in which mirroring could be implemented generically. Maybe I have to think about it a little bit more, but honestly I doubt that there is an easy way.

I'll put this issue to "discussion" to see what other users think about this requirement.
